src/librustc/middle/liveness.rs

```diff
@@ -1197,8 +1197,8 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
             }
 
             hir::ExprKind::Call(ref f, ref args) => {
-                // FIXME(canndrew): This is_never should really be an is_uninhabited
-                let succ = if self.tables.expr_ty(expr).is_never() {
+                let m = self.ir.tcx.hir.get_module_parent(expr.id);
+                let succ = if self.ir.tcx.is_ty_uninhabited_from(m, self.tables.expr_ty(expr)) {
```

Comment:
Note that this method uses a definition of uninhabitedness that also includes references and unions, which does not seem to match the desired semantics. Using it here in various stable functionality would commit to that behavior.